GI.JsonLoader/Core/ProgramManager.cs

```
@@ -0,0 +1,133 @@
+        private const string FileDataPath = @".\Data\";
+        
+        static ProgramManager()
+        {
+            ProgramManagerInstance = new ProgramManager();
+        }
+
+        internal ProgramManager() { }
+
+        public GIFile SelectedFile { get; private set; }
+
+        public bool TrySelectFile(string fileName)
+        {
+            string filePath = $"{FileDataPath}{fileName}";
+            if (!File.Exists(filePath))
+                return false;
+
+            try
+            {
+                var fileContents = JArray.Parse(File.ReadAllText(filePath));
+                var fileType = GetType(fileName);
+                var items = fileContents.Select(f => JsonConvert.DeserializeObject(f.ToString(), fileType));
+
+                SelectedFile = new GIFile(fileName, fileType, items);
+                return true;
+            }
+            catch (JsonReaderException)
+            {
+                Console.WriteLine(
+                    $"Invalid json for file path: {filePath} \r\n Please double check your data file to ensure it is valid");
+                return false;
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Exception has occurred attempting to load file: {ex.Message}");
+                return false;
+            }
+        }
+
+        public IEnumerable<object> GetFileContents(string fieldToSearch, string searchString)
+        {
+            // XXX: Instead of doing heavy reflection magic, we'll get more performance at the cost of maintainability here. 
+            switch (SelectedFile.Type.Name)
+            {
+                case nameof(Rpg):
+                    return SearchForRpgContents(fieldToSearch, searchString);
+                case nameof(Followers):
+                    return SearchForFollowersContents(fieldToSearch, searchString);
+
+                default: // XXX: By design, this shouldn't happen, but we're exposing this method publicly 
```

Could probably just let this fall through and return a more lightweight response of null.

In that case, however, you are relying on the caller to do the work in terms of handling this. Or it just silently fails. lol.